data_model/src/predicate.rs

```diff
@@ -0,0 +1,412 @@
+//! Predicate-related logic. Should contain predicate-related `impl`s.
+#![allow(unused_imports)]
+use raw_predicate::*;
+use string::*;
+
+use super::*;
+use crate::{IdBox, Identifiable, IdentifiableBox, Name, Value};
+
+/// Predicate combinator enum.
+#[derive(
+    Debug, Clone, Serialize, Deserialize, Encode, Decode, IntoSchema, PartialEq, Eq, PartialOrd, Ord,
+)]
+pub enum PredicateBox {
```

Comment:
this module looks very interesting. Will we be able to replace expressions with this if it proves worthy? At first glance their functionality seems on par

Reply:
Basically, this is the prototype for the composable filters. The expressions API has some overlap which I intend to remove. Since we have WASM, some things are now redundant (in addition to being woefully inadequate).